### ISSUE TITLE
Support use of id attribute for XLFF-based translations

### ISSUE THREAD

**Issue by steveWinter** [Haehnchen/idea-php-symfony2-plugin#885]
First of all thanks for adding the XLFF capability recently! I do however wish to request an extension to this. Within an XLFF translation file there are multiple `<trans-unit />` nodes which contain the translation. The format is intended such that one can have a source and a target language.

In most implementations the unique identifier is an id attribute of the `<trans-unit />` node, for example:
```
<trans-unit id="label.hello">
    <source>Hello</source>
    <target>Bonjour</target>
</trans-unit>
```
This allows someone using translation software to see the unique ID, the source language string and then perform the translation into the target language.

Symfony is happy with this approach, and will use **either** the id attribute, **or** the source string (as a fall-back if there is no matching id). Unfortunately the validator doesn't - the result as that all my translations are seen as missing.

Any chance that the validator can behave in the same way as Symfony does and support the use of either the id attribute or the source string?

**Comment by Haehnchen** [comment]
@steveWinter as i see this in not supported by Symfony. I have implemented resname support of #913. But i did not found an `id` implementation. see also https://phraseapp.com/blog/posts/translate-symfony-3-apps/:

> Most other Frameworks supporting XLIFF use id attributes to map translation keys. Symfony doesn’t work this way, but you can use the resname attribute for defining translation keys.

feel free to correct me :)

**Comment by steveWinter** [comment]
You are (of course) quite correct - it is as you say using resname rather than id. Thanks for implementing that - look forward to a new build that will remove all those 'Translation missing' errors :->
